File: Java/Easy/Accepted/sort-integers-by-the-number-of-1-bits_3.java
class Solution {
    public int[] sortByBits(int[] arr) {
        
        int[] bits = new int[arr.length];
        for (int i = 0; i < arr.length; i++) {
            bits[i] = Integer.bitCount(arr[i]);
        }
        for (int i = 0; i < arr.length; i++) {
            for (int j = i + 1; j < arr.length; j++) {
                if (bits[i] > bits[j]) {
                    int temp = arr[i];
                    arr[i] = arr[j];
                    arr[j] = temp;
                    temp = bits[i];
                    bits[i] = bits[j];
                    bits[j] = temp;
                } else if (bits[i] == bits[j]) {
                    if (arr[i] > arr[j]) {
                        int temp = arr[i];
                        arr[i] = arr[j];
                        arr[j] = temp;
                        temp = bits[i];
                        bits[i] = bits[j];
                        bits[j] = temp;
                    }
                }
            }
        }
        return arr;
    }
}

File: C++/Easy/WrongAnswer/sort-integers-by-the-number-of-1-bits_3.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        
        std::sort(arr.begin(), arr.end(), [](int a, int b) {
            int a1 = 0, b1 = 0;
            while (a) {
                a1 += a & 1;
                a >>= 1;
            }
            while (b) {
                b1 += b & 1;
                b >>= 1;
            }
            if (a1 == b1) return a < b;
            return a1 < b1;
        });
        return arr;
    }
};